lib/pharos/command_options/tf_json.rb

```diff
@@ -31,18 +32,25 @@ def load_external_config(config_hash)
         def load_terraform(file, config)
           puts("==> Importing configuration from Terraform ...".green) if $stdout.tty?
 
-          tf_parser = Pharos::Terraform::JsonParser.new(File.read(file))
-          config['hosts'] ||= []
-          config['api'] ||= {}
-          config['addons'] ||= {}
-          config['hosts'].concat(tf_parser.hosts)
-          config['api'].merge!(tf_parser.api) if tf_parser.api
-          config['name'] ||= tf_parser.cluster_name if tf_parser.cluster_name
-          config['addons'].each do |name, conf|
-            if addon_config = tf_parser.addons[name]
-              conf.merge!(addon_config)
+          json = File.read(file)
+          tf_parser = Pharos::Terraform::JsonParser.new(json)
+          if tf_parser.valid?
+            config.deep_merge!(tf_parser.cluster)
```

Comment:
Here's the defaults:

```
        # @param other [Hash]
        # @param overwrite_arrays [Boolean] when encountering an array, replace the array with the new array
        # @param union_arrays [Boolean] when encountering an array, use Array#union to combine with the existing array
        # @param keep_existing [Boolean] prefer old value over new value
        # @param merge_nil_values [Boolean] overwrite an existing value with a nil value
        # @param merge_non_hash [Boolean] calls .merge on objects that respond to .merge
        def deep_merge(other, overwrite_arrays: true, union_arrays: false, keep_existing: false, merge_nil_values: false, merge_non_hash: false)
```

This will replace any existing arrays with ones from TF, is that intended?

Reply:
> This will replace any existing arrays with ones from TF, is that intended?

 Not completely.. we should merge top-level `hosts` 🤔